misc/bazel/cmake/setup.cmake

```
@@ -0,0 +1,21 @@
+option(BUILD_SHARED_LIBS "" 0)
+
+execute_process(COMMAND bazel info workspace OUTPUT_VARIABLE BAZEL_WORKSPACE COMMAND_ERROR_IS_FATAL ANY OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+execute_process(COMMAND bazel info output_base OUTPUT_VARIABLE BAZEL_OUTPUT_BASE COMMAND_ERROR_IS_FATAL ANY OUTPUT_STRIP_TRAILING_WHITESPACE)
+string(REPLACE "-" "_" BAZEL_EXEC_ROOT ${PROJECT_NAME})
+set(BAZEL_EXEC_ROOT ${BAZEL_OUTPUT_BASE}/execroot/${BAZEL_EXEC_ROOT})
+
+execute_process(COMMAND bazel query "kind(generate_cmake, //...)" OUTPUT_VARIABLE BAZEL_GENERATE_CMAKE_TARGETS COMMAND_ERROR_IS_FATAL ANY OUTPUT_STRIP_TRAILING_WHITESPACE)
+execute_process(COMMAND bazel build ${BAZEL_GENERATE_CMAKE_TARGETS} COMMAND_ERROR_IS_FATAL ANY)
+
+string(REPLACE "//" "" BAZEL_GENERATE_CMAKE_TARGETS "${BAZEL_GENERATE_CMAKE_TARGETS}")
+string(REPLACE ":" "/" BAZEL_GENERATE_CMAKE_TARGETS "${BAZEL_GENERATE_CMAKE_TARGETS}")
+
+foreach (target ${BAZEL_GENERATE_CMAKE_TARGETS})
+    include(${BAZEL_WORKSPACE}/bazel-bin/${target}.cmake)
+endforeach ()
+
+if (CMAKE_EXPORT_COMPILE_COMMANDS)
+    file(CREATE_LINK ${PROJECT_BINARY_DIR}/compile_commands.json ${PROJECT_SOURCE_DIR}/compile_commands.json)
```

```suggestion
    file(CREATE_LINK ${PROJECT_BINARY_DIR}/compile_commands.json ${PROJECT_SOURCE_DIR}/compile_commands.json SYMBOLIC)
```

I had to do this because, initially (at `.ninja` file generation time), there is no `compile_commands.json` file yet, so a hard link cannot be created. It is only after compiling the code that the `compile_commands.json` gets generated. In contrast, a symbolic link doesn't need its source file to exist as it is defined by a path.

Additionally, I have had to change VSCode's `"cmake.buildDirectory"` to something like `"${workspaceFolder}/cmake-build"`. Perhaps we should consider un-.gitignore-ing `/.vscode/settings.json` and ensuring it has useful settings. In any case, it might be worth noting down the steps needed to enable all of this, including setting the CMAKE_EXPORT_COMPILE_COMMANDS to ON and installing ninja, in a README file.

After doing this, and initiating a semmle-code build through the VSCode UI, browsing the C Extractor source has been magical ✨ 🎉 Haven't tried it on the swift extractor yet.

---

I didn't have the same luck when building the swift extractor using CMake. First of all, VSCode didn't recognize a CMake project at `semmle-code/ql/swift`, even if I `include_subdirectory()` the latter from `semmle-code/CMakeLists.txt`. Second, when running CMake manually from the command line, there is an error that indicates that some include paths may not be set correctly:
```
╭─d10c@Noras-MBP ~/src/semmle-code/ql/swift ‹81628f595c●› 
╰─$ cmake --no-warn-unused-cli -DCMAKE_EXPORT_COMPILE_COMMANDS:BOOL=TRUE -DCMAKE_BUILD_TYPE:STRING=Debug -S/Users/d10c/src/semmle-code/ql/swift -B/Users/d10c/src/semmle-code/ql/cmake-build -G Ninja                                 1 ↵
Not searching for unused variables given on the command line.
-- The C compiler identification is AppleClang 14.0.0.14000029
-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
Loading: 0 packages loaded
Loading: 8 packages loaded
Loading: 8 packages loaded
Loading: 
Loading: 0 packages loaded
Analyzing: target //swift/extractor:cmake (0 packages loaded, 0 targets configured)
Analyzing: target //swift/extractor:cmake (20 packages loaded, 242 targets configured)
INFO: Analyzed target //swift/extractor:cmake (21 packages loaded, 393 targets configured).
INFO: Found 1 target...
[0 / 2] [Prepa] BazelWorkspaceStatusAction stable-status.txt
Target //swift/extractor:cmake up-to-date:
  bazel-bin/swift/extractor/cmake.cmake
INFO: Elapsed time: 2.587s, Critical Path: 0.01s
INFO: 1 process: 1 internal.
INFO: Build completed successfully, 1 total action
INFO: Build completed successfully, 1 total action
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/d10c/src/semmle-code/ql/cmake-build
╭─d10c@Noras-MBP ~/src/semmle-code/ql/swift ‹81628f595c●› 
╰─$ cd ../cmake-build/ 
CMakeCache.txt        CMakeFiles            build.ninja           cmake_install.cmake   compile_commands.json
╭─d10c@Noras-MBP ~/src/semmle-code/ql/cmake-build ‹81628f595c●› 
╰─$ ninja      
[1/30] Building CXX object CMakeFiles/_swift_extractor_extractor.dir/extractor/TargetTrapFile.cpp.o
FAILED: CMakeFiles/_swift_extractor_extractor.dir/extractor/TargetTrapFile.cpp.o 
/usr/bin/clang++  -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/swift_prebuilt_darwin_x86_64/include -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/swift_prebuilt_darwin_x86_64/include -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/swift_prebuilt_darwin_x86_64 -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/swift_prebuilt_darwin_x86_64 -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/fishhook -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/fishhook -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/bazel_tools -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/bazel_tools -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/fishhook/. -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/external/fishhook/. -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_extractor.dir/extractor/TargetTrapFile.cpp.o -MF CMakeFiles/_swift_extractor_extractor.dir/extractor/TargetTrapFile.cpp.o.d -o CMakeFiles/_swift_extractor_extractor.dir/extractor/TargetTrapFile.cpp.o -c /Users/d10c/src/semmle-code/ql/swift/extractor/TargetTrapFile.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
/Users/d10c/src/semmle-code/ql/swift/extractor/TargetTrapFile.cpp:1:10: fatal error: 'swift/extractor/TargetTrapFile.h' file not found
#include "swift/extractor/TargetTrapFile.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[2/30] Building CXX object CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b3...24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapClasses.cpp.o
FAILED: CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapClasses.cpp.o 
/usr/bin/clang++  -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapClasses.cpp.o -MF CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapClasses.cpp.o.d -o CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapClasses.cpp.o -c /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapClasses.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapClasses.cpp:3:
/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/./TrapClasses.h:9:10: fatal error: 'swift/extractor/trap/TrapLabel.h' file not found
#include "swift/extractor/trap/TrapLabel.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[3/30] Building CXX object CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapClasses.cpp.o
FAILED: CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapClasses.cpp.o 
/usr/bin/clang++  -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapClasses.cpp.o -MF CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapClasses.cpp.o.d -o CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapClasses.cpp.o -c /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapClasses.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapClasses.cpp:3:
/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/./TrapClasses.h:9:10: fatal error: 'swift/extractor/trap/TrapLabel.h' file not found
#include "swift/extractor/trap/TrapLabel.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[4/30] Building CXX object CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b3...07d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapClasses.cpp.o
FAILED: CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapClasses.cpp.o 
/usr/bin/clang++  -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapClasses.cpp.o -MF CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapClasses.cpp.o.d -o CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapClasses.cpp.o -c /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapClasses.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapClasses.cpp:3:
/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/./TrapClasses.h:9:10: fatal error: 'swift/extractor/trap/TrapLabel.h' file not found
#include "swift/extractor/trap/TrapLabel.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[5/30] Building CXX object CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b3...07d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/stmt/TrapEntries.cpp.o
FAILED: CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/stmt/TrapEntries.cpp.o 
/usr/bin/clang++  -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/stmt/TrapEntries.cpp.o -MF CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/stmt/TrapEntries.cpp.o.d -o CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/stmt/TrapEntries.cpp.o -c /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/stmt/TrapEntries.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/stmt/TrapEntries.cpp:3:
/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/stmt/./TrapEntries.h:8:10: fatal error: 'swift/extractor/trap/TrapLabel.h' file not found
#include "swift/extractor/trap/TrapLabel.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[6/30] Building CXX object CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b3...07d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapEntries.cpp.o
FAILED: CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapEntries.cpp.o 
/usr/bin/clang++  -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapEntries.cpp.o -MF CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapEntries.cpp.o.d -o CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapEntries.cpp.o -c /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapEntries.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/TrapEntries.cpp:3:
/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/decl/./TrapEntries.h:8:10: fatal error: 'swift/extractor/trap/TrapLabel.h' file not found
#include "swift/extractor/trap/TrapLabel.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[7/30] Building CXX object CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapEntries.cpp.o
FAILED: CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapEntries.cpp.o 
/usr/bin/clang++  -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapEntries.cpp.o -MF CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapEntries.cpp.o.d -o CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapEntries.cpp.o -c /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapEntries.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/TrapEntries.cpp:3:
/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/./TrapEntries.h:8:10: fatal error: 'swift/extractor/trap/TrapLabel.h' file not found
#include "swift/extractor/trap/TrapLabel.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[8/30] Building CXX object CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b3...24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapEntries.cpp.o
FAILED: CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapEntries.cpp.o 
/usr/bin/clang++  -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapEntries.cpp.o -MF CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapEntries.cpp.o.d -o CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapEntries.cpp.o -c /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapEntries.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/TrapEntries.cpp:3:
/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/pattern/./TrapEntries.h:8:10: fatal error: 'swift/extractor/trap/TrapLabel.h' file not found
#include "swift/extractor/trap/TrapLabel.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[9/30] Building CXX object CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b3...07d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapClasses.cpp.o
FAILED: CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapClasses.cpp.o 
/usr/bin/clang++  -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapClasses.cpp.o -MF CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapClasses.cpp.o.d -o CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapClasses.cpp.o -c /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapClasses.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapClasses.cpp:3:
/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/./TrapClasses.h:9:10: fatal error: 'swift/extractor/trap/TrapLabel.h' file not found
#include "swift/extractor/trap/TrapLabel.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[10/30] Building CXX object CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b...07d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapEntries.cpp.o
FAILED: CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapEntries.cpp.o 
/usr/bin/clang++  -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapEntries.cpp.o -MF CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapEntries.cpp.o.d -o CMakeFiles/_swift_extractor_trap_trap.dir/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapEntries.cpp.o -c /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapEntries.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/TrapEntries.cpp:3:
/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/swift/extractor/trap/generated/expr/./TrapEntries.h:8:10: fatal error: 'swift/extractor/trap/TrapLabel.h' file not found
#include "swift/extractor/trap/TrapLabel.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[11/30] Building CXX object CMakeFiles/_swift_extractor_extractor.dir/extractor/main.cpp.o
FAILED: CMakeFiles/_swift_extractor_extractor.dir/extractor/main.cpp.o 
/usr/bin/clang++  -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/swift_prebuilt_darwin_x86_64/include -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/swift_prebuilt_darwin_x86_64/include -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/swift_prebuilt_darwin_x86_64 -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/swift_prebuilt_darwin_x86_64 -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/fishhook -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/fishhook -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/bazel_tools -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/bazel_tools -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/fishhook/. -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/external/fishhook/. -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_extractor.dir/extractor/main.cpp.o -MF CMakeFiles/_swift_extractor_extractor.dir/extractor/main.cpp.o.d -o CMakeFiles/_swift_extractor_extractor.dir/extractor/main.cpp.o -c /Users/d10c/src/semmle-code/ql/swift/extractor/main.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
/Users/d10c/src/semmle-code/ql/swift/extractor/main.cpp:12:10: fatal error: 'swift/extractor/SwiftExtractor.h' file not found
#include "swift/extractor/SwiftExtractor.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
[12/30] Building CXX object CMakeFiles/_swift_extractor_extractor.dir/extractor/SwiftExtractor.cpp.o
FAILED: CMakeFiles/_swift_extractor_extractor.dir/extractor/SwiftExtractor.cpp.o 
/usr/bin/clang++  -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/swift_prebuilt_darwin_x86_64/include -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/swift_prebuilt_darwin_x86_64/include -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/swift_prebuilt_darwin_x86_64 -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/swift_prebuilt_darwin_x86_64 -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/fishhook -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/fishhook -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/bazel_tools -I/Users/d10c/src/semmle-code/ql/bazel-out/darwin_arm64-fastbuild/bin/external/bazel_tools -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/external/fishhook/. -I/private/var/tmp/_bazel_d10c/7c944f6a6d8b02e0b31125d907d24a29/execroot/codeql/bazel-out/darwin_arm64-fastbuild/bin/external/fishhook/. -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.3.sdk -std=c++17 -fno-rtti -arch=x86_64 -std=c++17 -MD -MT CMakeFiles/_swift_extractor_extractor.dir/extractor/SwiftExtractor.cpp.o -MF CMakeFiles/_swift_extractor_extractor.dir/extractor/SwiftExtractor.cpp.o.d -o CMakeFiles/_swift_extractor_extractor.dir/extractor/SwiftExtractor.cpp.o -c /Users/d10c/src/semmle-code/ql/swift/extractor/SwiftExtractor.cpp
clang: warning: argument unused during compilation: '-arch=x86_64' [-Wunused-command-line-argument]
In file included from /Users/d10c/src/semmle-code/ql/swift/extractor/SwiftExtractor.cpp:1:
/Users/d10c/src/semmle-code/ql/swift/extractor/SwiftExtractor.h:3:10: fatal error: 'swift/extractor/SwiftExtractorConfiguration.h' file not found
#include "swift/extractor/SwiftExtractorConfiguration.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
```

---

thanks for that, I will look into it now :slightly_smiling_face: 

About the symbolic link, good catch! I actually want to also test the link creation (symbolic or otherwise) on Windows. If this or something similar works on windows, then it might make sense to just enable `CMAKE_EXPORT_COMPILE_COMMANDS` by default so it works out of the box.

And yes, I should have mentioned the default build directory for VSCode does not work on our internal repo.

---

the macOS compilation issue is resolved now. Also, it is not necessary any more to set `CMAKE_EXPORT_COMPILE_COMMANDS`, it is on by default, with just an opt-out for Windows users who did not turn on developer mode (without which you can't create symbolic links as a normal user).